### PR TITLE
Add rubygem-foreman_probing_core package

### PR DESCRIPTION
**Pull request by dgoetz**
Add also rubygem-ruby-nmap and rubygem-rprogram package as dependency.
This is a Smart Proxy Dynflow Core plugin so I hope I used the correct template as start.

For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.19
* [ ] 1.18
* [ ] 1.17

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: packages/plugins/rubygem-foreman_probing_core/rubygem-foreman_probing_core.spec [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Generated from foreman_probing_core-0.0.1.gem by gem2rpm -*- rpm-spec -*-
+# template: scl
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+%global smart_proxy_dynflow_bundlerd_dir %{_datadir}/smart_proxy_dynflow_core/bundler.d
```
[the remark this answers is not in the document]
Comment (adamruzicka) [reply packages/plugins/rubygem-foreman_probing_core/rubygem-foreman_probing_core.spec]: This should go into `smart-proxy-probing` as @ekohl said.

[the remark this answers is not in the document]
Comment (dgoetz) [reply packages/plugins/rubygem-foreman_probing_core/rubygem-foreman_probing_core.spec]: This was taken from the [README](https://github.com/adamruzicka/foreman_probing#installing-the-smart-proxy-dynflow-core-plugin), so I am not sure how I should do the change. Would it be correct to add the `foreman_probing_core` as dependency to `smart-proxy-probing` and include the `bundler.d` parts in `smart-proxy-probing`?

[the remark this answers is not in the document]
Comment (adamruzicka) [reply packages/plugins/rubygem-foreman_probing_core/rubygem-foreman_probing_core.spec]: The readme covers development installation, in production we do things a bit differently.

> Would it be correct to add the foreman_probing_core as dependency to smart-proxy-probing and include the bundler.d parts in smart-proxy-probing?

That's the way we do it in REX, I'd stick with it.

**Comment by dgoetz** [comment]
I have changed the spec like requested. 

Is the additional provides in foreman_probing_core correct or should I better change the require in smart-proxy-probing? Without it provides and requires would not match one package being scl and the other non-scl.

**Comment by ehelms** [comment]
This needs a rebase please

**Comment by ekohl** [comment]
@evgeni any idea why this is failing? it looks related to your change in the job.

**Comment by evgeni** [comment]
@ekohl actually, yes ;)
It's failing because I was stupid ;)
PR incoming :)

**Comment by evgeni** [comment]
@ekohl https://github.com/theforeman/foreman-infra/pull/931 should fix it

**Comment by evgeni** [comment]
[test rpm]

**Comment by evgeni** [comment]
[test rpm]

**Comment by evgeni** [comment]
```
% koji add-pkg foreman-plugins-nightly-rhel7 tfm-rubygem-foreman_probing_core --owner=kojiadm
Adding 1 packages to tag foreman-plugins-nightly-rhel7
% koji add-pkg foreman-plugins-nightly-rhel7 tfm-rubygem-rprogram --owner=kojiadm            
Adding 1 packages to tag foreman-plugins-nightly-rhel7
% koji add-pkg foreman-plugins-nightly-rhel7 tfm-rubygem-ruby-nmap --owner=kojiadm 
Adding 1 packages to tag foreman-plugins-nightly-rhel7
```

**Pull request merged**